### ISSUE TITLE
sql: add builtins for PTS creation

### ISSUE THREAD

**Issue by jayshrivastava** [cockroachdb/cockroach#113493]
See https://github.com/cockroachlabs/support/issues/2697

When someone forgets to set `protect_data_from_gc_on_pause` and their changefeed is behind, they cannot alter it because altering requires pausing and pausing might result in the data being gc'd. It would be nice to have a builtin to create a PTS record to hold onto the data while we cancel and recreate the changefeed with the proper options. 

Sometimes we have bugs with altering and creating a new changefeed makes more sense (https://github.com/cockroachdb/cockroach/issues/113491).

By the way, you cannot just create a new changefeed job with a cursor equal to the highwater of the old one because the PTS record is not created with the cursor timetsamp (https://github.com/cockroachdb/cockroach/issues/113492).

**Comment by jayshrivastava** [comment]
Should we just get rid of `ALTER CHANGEFEED`? What if we created a new changefeed job instead and used some mechanism to transfer the PTS records.

**Comment by miretskiy** [comment]
I don't think we should get rid of alter changefeed -- if anything, ALTER is a well established UX.
